virtual_ecosystem/models/soil/env_factors.py

```diff
@@ -405,3 +407,30 @@ def calculate_leaching_rate(
     """
 
     return solubility_coefficient * solute_density * vertical_flow_rate / soil_moisture
+
+
+def find_total_soil_moisture_for_microbially_active_depth(
+    soil_moistures: DataArray,
+    layer_structure: LayerStructure,
+) -> NDArray[np.float32]:
+    """Find total soil moisture for the microbially active depth.
+
+    The proportion of each soil layer that lies within the microbially active zone is
+    first found. The soil moisture for each layer is then multiplied by this proportion
+    and summed, to find the total soil moisture in the microbially active zone.
+
+    Args:
+        soil_moistures: Soil moistures across all soil layers [mm]
+        layer_structure: The LayerStructure instance for the simulation.
+
+    Returns:
+        The total soil moisture in the microbially active depth [mm]
+    """
+
+    # Find the fraction of each layer that lies within the microbially active zone
+    layer_weights = (
+        layer_structure.soil_layer_active_thickness
+        / layer_structure.soil_layer_thickness
```

Comment:
I _think_ you might need to use the soil layer depth instead of the soil layer thickness here. If both layers are 1m thick, the depth of the first on is 1m, the depth of the second is 2m. if your microbial depth is then 1.2 m, you want all of the first layer but only 0.2 of the second layer, right? At the moment it looks like it always starts from zero?

Reply:
Okay, I'll add a note explaining what `layer_structure.soil_layer_active_thickness` is as that isn't really clear from the name alone